config_utilities/test/tests/factory.cpp

```diff
@@ -218,25 +218,42 @@ TEST(Factory, moduleNameConflicts) {
   const auto registration9 =
       config::Registration<TemplatedBase<float>, TemplatedDerived<float, float>>("float_derived");
   const std::string expected = R"""(Modules registered to factories: {
+  config::internal::ConfigWrapper(): {
+    'AddString' (config::test::AddString::Config),
+    'Derived2' (config::test::Derived2::Config),
+    'Derived2A' (config::test::Derived2A::Config),
+    'DerivedC' (config::test::DerivedC::Config),
+    'DerivedD' (config::test::DerivedD::Config),
+    'repeating' (config::test::RepeatingTalker::Config),
+  },
   config::internal::Formatter(): {
     'asl' (config::internal::AslFormatter),
   },
   config::internal::Logger(): {
     'stdout' (config::internal::StdoutLogger),
+    'test_logger' (config::test::TestLogger),
+  },
+  config::test::Base(_, int): {
```

Comment:
If I understand the `_` are skip-first arg registrations? It's not very clear to me how to read these (the initial idea of the design was to show the user how to invoke all these types/factories. Not clear to me how I would go about this for a Object(_)? Open to changes though 🙂

Reply:
Yeah, sorry meant to check with you about what you wanted here and then forgot. The `_` is to mark that these are actually `ObjectWithConfig` registrations. The reason for the skip first is because otherwise you get `YAML::Node` as the first arg which looked ugly and was confusing (though now that I think about it, it think `ObjectFactory::create` would dispatch properly to the `ObjectWithConfigFactory` version if you called it with a `YAML::Node` arg first). I also tried `*` as an option (i.e., as a `Any` type), but didn't think it was as intuitive (the `_` is meant to be like `std::placeholder::_1`). Another option might be `ConfigT` but I was a little nervous about that being confused for an instantiable type.  I could also drop the extra arg entirely from the signature, but then you get two identical signatures with different modules next to each other which is also confusing